src/coreclr/jit/emitarm64.cpp

```diff
@@ -10729,6 +10744,24 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
             code |= insEncodeReg_Rt(id->idReg2());             // ttttt
             code |= insEncodeReg_Rn(id->idReg3());             // nnnnn
             dst += emitOutput_Instr(dst, code);
+
+            // Some instructions with this encoding return their result in the
+            // second operand register instead of the first so we special case
+            // the GC update here and skip the common path down below.
+            if (emitInsDestIsOp2(ins))
+            {
+                if (id->idGCref() != GCT_NONE)
+                {
+                    emitGCregLiveUpd(id->idGCref(), id->idReg2(), dst);
+                }
+                else
+                {
+                    emitGCregDeadUpd(id->idReg2(), dst);
+                }
+
+                goto SKIP_GC_UPDATE;
```

Comment:
Question - Natural place to do this is down below before `SKIP_GC_UPDATE`, but you prefer to update it here, so most of the other formats don't have to execute this extra ` if (emitInsDestIsOp2(ins))`?

Reply:
Yes, exactly, I did first try to check it down there but it had some TP cost.